lib/system.nim

```diff
@@ -2050,20 +2050,56 @@ template unlikely*(val: bool): bool =
       unlikelyProc(val)
 
 const
+  ## TODO: in the next csources release:
+  ## - create a tuple for the compiler and stdlib version
+  ## - create a string for the both of these versions
+  ## - compiler version must be a magic, hardcoded by the compiler
+  ## - standard library version must be hardcoded in the stdlib source
+  ## - implement comparison operators and such niceities for versions
+  ## - toss NimVersion and all that
+  CompilerVersionMajor* {.intdefine.}: int = 1
+    ## major version number for the current compiler.
+    ## TODO: change to `0` in next csources
+    ## 
+    ## .. code-block:: Nim
+    ##   when CompilerVersionMajor > 0: echo "stability cargo culting"
+  CompilerVersionMinor* {.intdefine.}: int = 6
+    ## minor version number for the current compiler.
+    ## TODO: change to `1` in next csources
+  CompilerVersionPatch* {.intdefine.}: int = 0
+    ## patch version number for the current compiler.
+    ## TODO: change to `0` in next csources
+
+  StdlibMajor*: int = 1
+    ## standard library major version
+    ## TODO: change to `0` in next csources
+    ## 
+    ## Example:
+    ## 
+    ## .. code-block:: Nim
+    ##   when (StdlibMajor, StdlibMinor, StdlibPatch) >= (1, 3, 1): discard
+  StdlibMinor*: int = 6
+    ## standard library minor version
+    ## TODO: change to `1` in next csources
+  StdlibPatch*: int = 0
+    ## standard library patch version
+    ## TODO: change to `0` in next csources
+
+  # hardcoded in config/config.nims
```

Comment:
I dropped the code that hardcodes them, but might bring that back in a subsequent commit.